packages/n/nodejs/xmake.lua

```diff
@@ -0,0 +1,15 @@
+package("nodejs")
+
+    set_homepage("https://nodejs.org/")
+    set_description("Cross-platform JavaScript runtime environment.")
+
+    on_fetch("linux", function (package, opt)
+        if opt.system then
+            import("lib.detect.find_path")
+
+            local headers_path = find_path("node.h", { "/usr/**", "/usr/local/**"})
```

Comment:
We should not return `/usr/include` path, it will break for gcc.

please use find_package.

https://github.com/xmake-io/xmake-repo/blob/12301519c42acd15ba290de1b6cc00cae61d1e00/packages/l/libiconv/xmake.lua#L28

Reply:
ok no problem, can you explain why ? it seems to work fine for me with gcc-13

Reply:
oh is it because the linux headers have file named `node.h` too ?

Reply:
i was only able to find the package with `package:find_package("apt::nodejs")` (i'm on ubuntu). `system` and `pkgconfig` did not work.  
is it ok if i use this: `return package:find_package("apt::nodejs") or package:find_package("pacman::nodejs")` ?

Reply:
`package:find_tool("nodejs")` seems to work too, not sure which one is preferable

Reply:
i'm using local headers_path = find_path("node_api.h", { "/usr/**", "/usr/local/**"}) in my local package for now and everything seems to be working fine.

Reply:
it will pass `-isystem /usr/include` or `-isystem /usr/local/include` to gcc, and it will break gcc.

This is a known issue with gcc. If `node_api.h` is in /usr/include and /usr/local/include, gcc will find it even if you don't add `-isystem /usr/include`.

https://github.com/xmake-io/xmake/issues/4596#issuecomment-1886711974
https://github.com/msys2/MINGW-packages/issues/10761
https://github.com/pcb2gcode/pcb2gcode/issues/587

The `package:find_package("system::xx")` automatically handles this and avoids adding unnecessary `-system /usr/include`. like this https://github.com/xmake-io/xmake-repo/blob/12301519c42acd15ba290de1b6cc00cae61d1e00/packages/l/libiconv/xmake.lua#L28

Reply:
Ok thank you for the explanation.

The headers are located in `/usr/include/node/` so that's why it is working for me.

I still not sure why `package:find_package("system:nodejs")` doesn't find anything but I dug a little more and according to https://github.com/nodejs/node-addon-api/issues/882, there is no way to find the path to the headers that come packaged with `nodejs` in a platform agnostic way.

However, this issue indirectly links to https://github.com/nodejs/node-api-headers which seems to contain exactly what I need for `node-addon-api`. So I think I should probably create a package from this repo instead.

I'll wait for your feedback before creating yet another PR

Reply:
> So I think I should probably create a package from this repo instead.node-addon-api

you can try.

> The headers are located in so that's why it is working for me./usr/include/node/
> I still not sure why doesn't find anything but I dug a little more and according to https://github.com/nodejs/node-addon-api/issues/882, there is no way to find the path to the headers that come packaged with in a platform agnostic way.package:find_package("system:nodejs")nodejs

try

```lua
package:find_package("system:nodejs", {includes = "node/node_api.h"})
```

Reply:
> try
> 
> ```lua
> package:find_package("system:nodejs", {includes = "node/node_api.h"})
> ```

Still nothing :/

I have made a `node-api-headers` package that works perfectly for me. It as the advantage of being way lighter as it doesn't contain the whole Node runtime + ecosystem but just the headers that I need for my c++ addon. I'll make a PR right now.